src/app/hooks/api/useFavorites.js

```diff
@@ -0,0 +1,27 @@
+import { useState } from "react";
+
+export const useFavorites = () => {
+  const [favorite, setFavorite] = useState(() => {
+    if (
+      localStorage.getItem("favorite") !== null &&
+      localStorage.getItem("favorite") !== "undefined"
```

Comment:
@Sabian329 well, the 7th line checks if `"favorite" !== "undefined"`; `"undefined"` being a non-empty string here, meaning it would convert to truthy value, so your proposed solution isn't the same. However I also wonder if said `"undefined"` as a non-empty string is really desired here in such form, but I believe it might be.

Reply:
You're right. It is not the same solution. My bad, the best way to check if a variable is defined and not null is @Kojot999 way. 

https://stackoverflow.com/a/13392335

Reply:
You're both wrong. Most sexy solution is achievable thanks to **nullish coalescing operator "??"**.

Do it just this way:

`
localStorage.getItem("favorite") ?? false
`

The nullish coalescing operator (??) is a logical operator that returns its right-hand side operand when its left-hand side operand is **null or undefined**, and otherwise returns its left-hand side operand.

Reply:
Additionally. If u check docs of LocalStorage API you can see that if key doesn't exist it returns null not undefined, so there is no reason to check if value is undefined.

https://developer.mozilla.org/en-US/docs/Web/API/Storage/getItem

Reply:
**Evidence NO.1**
![image](https://user-images.githubusercontent.com/58563491/176429816-f123ff17-273e-4aa7-ad14-b3deb0076fc7.png)

Reply:
Its done Ernest way